node/src/components/contract_runtime.rs

```diff
@@ -392,6 +392,10 @@ where
                     let start = Instant::now();
                     let result = engine_state
                         .put_trie_and_find_missing_descendant_trie_keys(correlation_id, &*trie);
+                    // PERF: this *could* be called only periodically.
+                    if let Err(lmdb_error) = engine_state.flush_environment() {
+                        error!(?lmdb_error, "error flushing lmdb environment");
```

Comment:
I think we're treating an error while flushing as fatal at the other callsite.  That would be appropriate here too wouldn't it?

Reply:
Nice catch. I'll do that now.

Reply:
Done